algorithms/src/merkle_tree/merkle_path.rs

```diff
@@ -37,7 +37,7 @@ impl<P: MerkleParameters> MerklePath<P> {
     pub fn verify<L: ToBytes>(&self, root_hash: &MerkleTreeDigest<P>, leaf: &L) -> Result<bool, MerkleError> {
         // Check that the given leaf matches the leaf in the membership proof.
         if !self.path.is_empty() {
-            let hash_input_size_in_bytes = (P::H::INPUT_SIZE_BITS / 8) * 2;
+            let hash_input_size_in_bytes = P::H::INPUT_SIZE_BITS / 8;
```

Comment:
Previously, the Merkle tree never panics, because it always reserves twice the input size. 

So there is no panicking if the parameters is down by a half -> just somewhere in the code, the second half disappears.